plugins/serializers/carbon2/carbon2.go

```diff
@@ -0,0 +1,66 @@
+package carbon2
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"strconv"
+	"strings"
+)
+
+type serializer struct {
+}
+
+func NewSerializer() (*serializer, error) {
+	s := &serializer{}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+	return []byte(s.createObject(metric)), nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+	var batch bytes.Buffer
+	for _, metric := range metrics {
+		batch.WriteString(s.createObject(metric))
+	}
+	return []byte(batch.String()), nil
+}
+
+func (s *serializer) createObject(metric telegraf.Metric) string {
+	var m bytes.Buffer
+	for fieldName, fieldValue := range metric.Fields() {
+		if isNumeric(fieldValue) {
+			m.WriteString("metric=")
+			m.WriteString(strings.Replace(metric.Name(), " ", "_", -1))
+			m.WriteString(" field=")
+			m.WriteString(strings.Replace(fieldName, " ", "_", -1))
+			m.WriteString(" ")
+			for k, v := range metric.Tags() {
```

Comment:
Use TagList() instead, this will give you the tags in sorted order, since the specifications says:

> For intrinsic tags, pick an order and stick with it, otherwise the metric id will change

This will also more performance because it doesn't require a map to be created for each field.